src/librbd/api/Trash.cc

```diff
@@ -474,7 +474,7 @@ int Trash<I>::purge(IoCtx& io_ctx, time_t expire_ts,
 
         for (const auto &it : img->second) {
           auto ictx = new I("", it, nullptr, io_ctx, false);
-          r = ictx->state->open(OPEN_FLAG_SKIP_OPEN_PARENT);
+          r = ictx->state->open(OPEN_FLAG_SKIP_OPEN_PARENT | OPEN_FLAG_REMOVE_IMAGE);
```

Comment:
My test didn't go through this code path. I guess "rbd trash purge" will run to here, so I add here too. I don't know how to how to trigger it. Please evaluate the changes or give some test advice, thanks.

Reply:
@CongMinYin  You have to run `rbd trash purge` with `--threshold` argument to trigger it. Something like:
```
rbd trash move $pool/$image
rbd trash purge $pool --threshold 0.01
```
Then it will open every trash image to calculate its usage. But I think you need `OPEN_FLAG_REMOVE_IMAGE` flag here, as it is not removing yet.

Reply:
> @CongMinYin You have to run `rbd trash purge` with `--threshold` argument to trigger it. Something like:
> 
> ```
> rbd trash move $pool/$image
> rbd trash purge $pool --threshold 0.01
> ```
> 
> Then it will open every trash image to calculate its usage. But I think you don't need `OPEN_FLAG_REMOVE_IMAGE` flag here, as it is not removing yet.

@trociny I ran these 2 commands, but it didn't trigger here. I have removed the flag here because I'm not sure how this will be called.